ngraph/frontend/paddlepaddle/src/model.cpp

```diff
@@ -228,6 +235,77 @@ namespace ngraph
 #endif
         } // namespace pdpd
 
+        std::vector<std::shared_ptr<OpPlacePDPD>>
+            InputModelPDPD::InputModelPDPDImpl::returnOpPlaces()
+        {
+            if (m_graph_changed)
+            {
+                determine_cut_nodes();
+                m_graph_changed = false;
+            }
+            return m_op_places;
+        }
+
+        void InputModelPDPD::InputModelPDPDImpl::determine_cut_nodes()
+        {
```

Comment:
Hard to review without tests. Did you test case when user cuts some input which is important to produce some output?

Reply:
I will add such test